public/app/plugins/datasource/prometheus/components/VariableQueryEditor.tsx

```diff
@@ -120,7 +123,11 @@ export const PromVariableQueryEditor = ({ onChange, query, datasource }: Props)
 
     const queryString = migrateVariableEditorBackToVariableSupport(updatedVar);
 
-    const lblFltrs = updLabelFilters ? updLabelFilters : labelFilters;
+    let lblFltrs = updLabelFilters ? updLabelFilters : labelFilters;
+
+    if (labelNamesMatch) {
+      lblFltrs = [...lblFltrs, { label: '__name__', op: '=~', value: labelNamesMatch }];
```

Comment:
What was the purpose of adding the labelNamesMatch here? Do we have to do this? When we do this, we are storing the label filter for labelNamesmatch every blur event in the variable object and we get this side effect

![Screenshot 2023-06-21 at 3 30 36 PM](https://github.com/grafana/grafana/assets/25674746/438dbe36-f249-49cd-ae23-3b4552b54bad)

![Screenshot 2023-06-21 at 3 30 46 PM](https://github.com/grafana/grafana/assets/25674746/edfad090-eb86-489a-9231-db42c267fcd8)

Reply:
Good catch! Fixed and pushed

Reply:
I don't know what happened here lol